Source/WebKit/UIProcess/glib/DisplayLinkGLib.cpp

```diff
@@ -35,10 +37,28 @@ void DisplayLink::platformInitialize()
     // FIXME: We can get here with displayID == 0 (webkit.org/b/212120), in which case DisplayVBlankMonitor defaults to the main screen.
     m_vblankMonitor = DisplayVBlankMonitor::create(m_displayID);
     m_vblankMonitor->setHandler([this] {
-        notifyObserversDisplayDidRefresh();
+        m_fpsThrottleCallCounter++;
+        if (m_fpsThrottleCallCounter >= m_fpsThrottleRatio) {
+            notifyObserversDisplayDidRefresh();
+            m_fpsThrottleCallCounter = 0;
+        }
     });
 
     m_displayNominalFramesPerSecond = m_vblankMonitor->refreshRate();
+    if (const char* envString = getenv("WEBKIT_PROCESS_DISPLAY_LINK_THROTTLE_FPS")) {
+        if (auto newValue = parseInteger<unsigned>(StringView::fromLatin1(envString))) {
+            unsigned refreshRate = m_vblankMonitor->refreshRate();
+            // The throttle FPS must be a factor of the real display refresh rate
+            if (!(refreshRate % *newValue) && 0 < *newValue < refreshRate)
```

Comment:
The `lowerBound < value < upperBound` construct is a Pythonishm 🐍. It should be something like the following:


```suggestion
            if (!(refreshRate % *newValue) && *newValue > 0 && *newValue < refreshRate)
```